IDE/STM32Cube/wolfssl_example.c

```diff
@@ -1468,25 +1469,264 @@ static void PrintMemStats(void)
+    WOLFSSL* ssl = NULL;
+    byte echoBuffer[100];
+    tls13_buf tbuf;
+
+    XMEMSET(&tbuf, 0, sizeof(tbuf));
+    /* TODO: Flush any data in the RX buffer */
+
+    ctx = wolfSSL_CTX_new(wolfTLSv1_3_server_method());
+    if (ctx == NULL) {
+        printf("Error creating WOLFSSL_CTX\n");
+        goto done;
+    }
+
+    /* Register wolfSSL send/recv callbacks */
+    wolfSSL_CTX_SetIOSend(ctx, uartIOTx);
+    wolfSSL_CTX_SetIORecv(ctx, uartIORx);
+
+    ret = wolfSSL_CTX_use_PrivateKey_buffer(ctx, ecc_key_der_256,
+        sizeof_ecc_key_der_256, WOLFSSL_FILETYPE_ASN1);
+    if (ret != WOLFSSL_SUCCESS) {
+        printf("error loading server private key\n");
+        goto done;
+    }
+
+    ret = wolfSSL_CTX_use_certificate_buffer(ctx, serv_ecc_der_256,
+        sizeof_serv_ecc_der_256, WOLFSSL_FILETYPE_ASN1);
+    if (ret != WOLFSSL_SUCCESS) {
+        printf("error loading server certificate\n");
+        goto done;
+    }
+
+    ssl = wolfSSL_new(ctx);
+    if (ssl == NULL) {
+        printf("Error creating WOLFSSL\n");
+        goto done;
+    }
+
+    wolfSSL_SetIOReadCtx(ssl, &tbuf);
+
+    printf("Waiting for client\n");
+    do {
+        ret = wolfSSL_accept(ssl);
+        err = wolfSSL_get_error(ssl, ret);
+    } while (err == WOLFSSL_ERROR_WANT_READ || err == WOLFSSL_ERROR_WANT_WRITE);
+    if (ret != WOLFSSL_SUCCESS) {
+        printf("TLS accept error %d\n", err);
+        goto done;
+    }
+    printf("TLS Accept handshake done\n");
+
+    /* Waiting for data to echo */
+    XMEMSET(echoBuffer, 0, sizeof(echoBuffer));
+    do {
+        ret = wolfSSL_read(ssl, echoBuffer, sizeof(echoBuffer)-1);
+        err = wolfSSL_get_error(ssl, ret);
+    } while (err == WOLFSSL_ERROR_WANT_READ || err == WOLFSSL_ERROR_WANT_WRITE);
+    printf("Read (%d): %s\n", err, echoBuffer);
+
+    do {
+        ret = wolfSSL_write(ssl, echoBuffer, XSTRLEN((char*)echoBuffer));
+        err = wolfSSL_get_error(ssl, ret);
+    } while (err == WOLFSSL_ERROR_WANT_READ || err == WOLFSSL_ERROR_WANT_WRITE);
+    printf("Sent (%d): %s\n", err, echoBuffer);
+
+    ret = 0; /* Success */
+
+done:
+    if (ssl) {
+        wolfSSL_shutdown(ssl);
+        wolfSSL_free(ssl);
+    }
+    if (ctx) {
+        wolfSSL_CTX_free(ctx);
+    }
+
+    return ret;
 }
+
+static int tls13_uart_client(void)
+{
+    int ret = -1, err;
+    WOLFSSL_CTX* ctx = NULL;
+    WOLFSSL* ssl = NULL;
+    const char testStr[] = "Testing 1, 2 and 3\r\n";
+    byte readBuf[100];
+    tls13_buf tbuf;
+
+    XMEMSET(&tbuf, 0, sizeof(tbuf));
+    /* TODO: Flush any data in the RX buffer */
+
+    ctx = wolfSSL_CTX_new(wolfTLSv1_3_client_method());
+    if (ctx == NULL) {
+        printf("Error creating WOLFSSL_CTX\n");
+        goto done;
+    }
+
+    /* Register wolfSSL send/recv callbacks */
+    wolfSSL_CTX_SetIOSend(ctx, uartIOTx);
+    wolfSSL_CTX_SetIORecv(ctx, uartIORx);
+
+    /* For testing disable peer cert verification */
+    wolfSSL_CTX_set_verify(ctx, SSL_VERIFY_NONE, NULL);
```

Comment:
Let's turn on the server side certificate checking. Easy to do by loading the right server cert or ca.

Reply:
Sure! I was doing it before but changed it back.

Reply:
huh....can't figure out which is the right CA Cert.  This will be a TODO for a while.

Reply:
Ah....Actually. I think its a good idea to keep this disabled.  You see the user is going to get the STM code from cube and then on the PC side they are going to get the code from who knows where.  If the code is out of sync (which is very likely) then there will not have matching certificate chains.